nodejs/week2/app.js

```diff
@@ -0,0 +1,75 @@
+const express = require('express');
+const app = express();
+const port = process.env.PORT || 3000;
+const documents = require('./documents.json');
+
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send(documents);
+});
+
+app.get('/search', (req, res) => {
+  const query = req.query.q;
+
+  if (query) {
+    const result = documents.filter((element) =>
+      Object.values(element).some((item) => String(item).includes(query))
+    );
+    if (result.length === 0) {
+      res.status(404).json({ error: 'Record not found' });
```

Comment:
When a request responds with an empty object/array, it is still a successful request and should respond with status 200, and a message instead of an error.